api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerLaunchTest.java

```diff
@@ -328,6 +347,19 @@ public void workerRunShouldUseParentRunCloudRegionWithParentRunIdPassedAsParamet
         assertThat(pipelineRun.getInstance().getCloudRegionId(), is(NON_DEFAULT_REGION_ID));
     }
 
+    @Test
+    public void shouldLoadRunsActivityStats() {
+        doReturn(Arrays.asList(getPipelineRun(ID, TEST_USER), getPipelineRun(ID_2, TEST_USER)))
+                .when(pipelineRunDao).loadPipelineRunsActiveInPeriod(eq(TEST_PERIOD), eq(TEST_PERIOD_18));
+        doReturn(getStatusMap()).when(runStatusManager).loadRunStatus(anyListOf(Long.class));
+
+        pipelineRunManager.loadRunsActivityStats(TEST_PERIOD, TEST_PERIOD_18).stream()
+                .collect(Collectors.toMap(BaseEntity::getId, Function.identity()));
+
+        verify(pipelineRunDao).loadPipelineRunsActiveInPeriod(any(LocalDateTime.class), any(LocalDateTime.class));
+        verify(runStatusManager).loadRunStatus(anyListOf(Long.class));
+    }
```

Comment:
If you look at this test after all the changes applied it doesn't look like it tests anything what `pipelineRunManager.loadRunsActivityStats` really does. At this point there is no sense using `getStatusMap()` you could replace it with `Collections.emptyMap()` and the test will still pass.

The whole idea behind `pipelineRunManager.loadRunsActivityStats` is the following:
1. It loads active runs.
2. It loads statuses for found active runs.
3. It merges loaded runs with loaded statuses.

As a result in the test you should mock 1 and 2 points and verify the 3. You should verify that the runs returned by the target method have associated run statuses. You could even skip `verify` calls if you check the returned runs instead. That was the idea behind the original test and we don't have to change it.

The test should look something like
```java
mockRuns
mockStatuses

loadRunsWithStatuses

assertLoadedRunsHaveAssociatedStatuses
```

Reply:
Resolved!